maestrowf/maestro.py

```diff
@@ -334,6 +336,11 @@ def setup_argparser():
     run.add_argument("--dry", action="store_true", default=False,
                      help="Generate the directory structure and scripts for a "
                      "study but do not launch it. [Default: %(default)s]")
+    run.add_argument("--draw", type=str, action="append", default=[],
```

Comment:
It might be worth gating the command line parameter addition behind a package check as seen here: https://stackoverflow.com/a/27361558

Reply:
Yeah, i saw that discussion when looking into best practices for this kind of feature disablement.  I'm a bit torn on doing it in function vs setting a file level variable at the top; file level might be more appropriate as I'm thinking the vis function needs splitting (particularly for future use in additional layers on top of maestro where you might want to return/serialize the networkx representation instead of rendering it on the spot).

As for gating the command line parameter, there is a hangup there which led me to doing it in the function: import time.  Doing a bunch of extra import tests in the main driver will most definitely slow things down, whereas when it's gated in the conductor as it is now the user never sees it.  Although it does delay the check and do a lot of work, potentially for nothing by delaying the import checks until right before they're needed..  Will have to dig some more on whether such checks can be done via alternate means to catch it early without always slowing it down.

Reply:
I'm not sure if the import checks really do take that long, might have to test that. But the test upfront let's the user know that the feature isn't enabled. By pushing it to the conductor, the user might not know they don't have it installed and then it silently fails. I think a simple import check for one of the three packages might suffice?

Reply:
Tweaked to have choices and updated help message:
```
  --draw {mpl,mpl-dot,graphml,graphml-dot}
                        Generate visualizations of study's DAG (specify --draw
                        multiple times with various args to draw multiple
                        formats). Supported formats: matplotlib png (mpl), dot
                        layout mpl (mpl-dot), graphviz dot file (dot), graphml
                        file (graphml), dot layout positioned graphml file
                        (graphml-dot)
```